bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Accumulation/AccumulatedSelection.cs

```diff
@@ -0,0 +1,194 @@
+        SequenceOrder = selection.SequenceOrder;
+        DescriptionHash = new(selection.DescriptionHash);
+        Value = new(Constants.ONE_MOD_P);
+        Commitment = new(Constants.ONE_MOD_P, Constants.ONE_MOD_P);
+    }
+
+    public AccumulatedSelection(
+        AccumulatedSelection other) : base(other)
+    {
+        ObjectId = other.ObjectId;
+        SequenceOrder = other.SequenceOrder;
+        DescriptionHash = new(other.DescriptionHash);
+        Value = new(other.Value);
+        if (other.Proof is not null)
+        {
+            Proof = new(other.Proof);
+        }
+        Commitment = new(other.Commitment);
+        Shares = other.Shares.ToDictionary(
+            x => x.Key,
+            x => new SelectionShare(x.Value));
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        List<Tuple<ElectionPublicKey, SelectionShare>> guardianShares,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        var lagrangeCoefficients = guardianShares.ComputeLagrangeCoefficients();
+
+        foreach (var (guardian, share) in guardianShares)
+        {
+            Accumulate(share, lagrangeCoefficients[guardian.GuardianId]);
+        }
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        List<Tuple<ElectionPublicKey, SelectionShare>> guardianShares,
+        Dictionary<string, LagrangeCoefficient> lagrangeCoefficients,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        foreach (var (guardian, share) in guardianShares)
+        {
+            Accumulate(share, lagrangeCoefficients[guardian.GuardianId]);
+        }
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        SelectionShare share,
+        LagrangeCoefficient lagrangeCoefficient,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        var copy = new SelectionShare(share);
+        Shares.Add(share.GuardianId, copy);
+        Accumulate(
+            copy.Share,
+            copy.Commitment,
+            lagrangeCoefficient.Coefficient);
+    }
+
+    /// <summary>
+    /// Add a CP proof to the selection. The proof is generated by the guardians through a protocol
+    /// </summary>
+    public void AddProof(ChaumPedersenProof proof)
+    {
+        Proof = new(proof);
+    }
+
+    public bool IsValid()
+    {
+        // TODO: better validation
+        if (Shares.Count == 0)
+        {
+            return false;
+        }
+
+        if (Proof is null)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected override void DisposeManaged()
```

Comment:
for the `Decryption` project only I broke out the dispose into managed and un managed as it was originally intended. managed dispose dumps collections and wrapper objects and then the unmanaged does the native objects. Really, since we are using managed wrappers around the native objects, they could all just be in dispose managed and the internals of objects like ElementModP should handle the unamanaged portion, but this way theres a clear distinction between what is _actually_ a managed object and what is something that has a native component to it. idk if thats valuable but it's hwo i did it.